lightningd/lightningd.c

```diff
@@ -783,9 +783,13 @@ static struct io_plan *sigchld_rfd_in(struct io_conn *conn,
 	/* We don't actually care what we read, so we stuff things here. */
 	static u8 ignorebuf;
 	static size_t len;
+	pid_t pid;
 
-	/* Reap the plugins, since we otherwise ignore them. */
-	while (waitpid(-1, NULL, WNOHANG) != 0);
+	/* Reap any terminated child we can; for the rest we ignore these */
+	while ((pid = waitpid(-1, 0, WNOHANG)) != 0)
+		/* No child processes: this causes a hang sometimes, so break that */
+		if (pid == -1 && errno == ECHILD)
+			break;
 
```

Comment:
I don't think this is what you're seeing; I suspect the OOM killer is hitting us.

Reply:
yes, dropped it